Nurgissa/nodejs2024Q3-service
# [Task-7] Home Library Service - part 1

## Assignment details
[Home Library Service #1](https://github.com/AlreadyBored/nodejs-assignments/blob/main/assignments/rest-service/assignment.md)

## Disclaimer
Dear reviewer, in order to install dependencies of the project please use below command:
```
npm i --force   
```

### Why do I need to `force` install?
Due to conflict of the version of `@nest/swagger`, we need to force install.

## Feedback from the reviewers
It turns out with my last clean up commit, I broke the app and it is failing to run due to typecheck errors now. Below is the screenshot of the errors.
![image](https://github.com/user-attachments/assets/2eda10f0-fedb-4818-a1fa-a25e34793976)

Please remove the usage of `ArtistDto` and `AlbumDto` from entities as follows and start the application.
![image](https://github.com/user-attachments/assets/02f23b93-d02c-4c88-a779-8cdb3730fd38)
![image](https://github.com/user-attachments/assets/7965429c-6753-41d9-a5eb-936f980d5b30)

### How to run
- To start in development mode, type `npm run start:dev`
- To start in production mode, type `npm run start:prod`
After that:
- Execute API tests, type `npm run test`



## Submission details
I have implemented the assignment completely before deadline.

#### Result of running e2e tests
![image](https://github.com/user-attachments/assets/ed84510b-36fd-4e49-b692-61e0d10d3163)

#### Result of executing `npm run lint`
![image](https://github.com/user-attachments/assets/196f2677-7a8d-4938-b649-5cb8c161e8fe)

#### Swagger for home library service
![image](https://github.com/user-attachments/assets/cec854e6-5cbb-4020-a3c1-2213c0e8ecbe)


### What to test
- Run tests
- Review code
- Check .env file
